app/components/add-cron-job.js

```diff
@@ -1,45 +1,68 @@
 import Component from '@ember/component';
 import { inject as service } from '@ember/service';
-import { task } from 'ember-concurrency';
+import { computed } from '@ember/object';
+import { mapBy } from '@ember/object/computed';
 
 export default Component.extend({
   store: service(),
+  branchesFetcher: service(),
 
   classNames: ['form--cron'],
 
+  intervals: ['monthly', 'weekly', 'daily'],
+
+  options: ['Always run', 'Do not run if there has been a build in the last 24h'],
+
+  currentCronJobsBranches: mapBy('repository.cronJobs', 'branch.name'),
+
+  init() {
+    this.reset();
+    this.setProperties({
+      'branchesFetcher.repository': this.get('repository')
+    });
+    this.get('branchesFetcher.showSearchResults').perform();
+    this._super(...arguments);
+  },
+
   reset() {
     this.setProperties({
       selectedBranch: null,
-      selectedInterval: null,
+      selectedInterval: this.intervals.firstObject,
+      selectedOption: this.options[0],
       disable: null
     });
   },
 
-  save: task(function* () {
-    const store = this.get('store');
-    const repoId = this.get('branches.firstObject.repoId');
-    const branch = this.get('selectedBranch') || this.get('branches.firstObject');
-
-    const existingCrons = yield store.filter('cron', { repository_id: repoId }, (c) => {
-      c.get('branch.repoId') === repoId && c.get('branch.name') === branch.get('name');
-    });
+  comparedBranches: computed('currentCronJobsBranches', 'branchesFetcher.searchResults', function () {
+    let branchNames = this.get('currentCronJobsBranches');
+    let branches = this.get('branchesFetcher.searchResults');
 
-    if (existingCrons.get('firstObject')) {
-      store.unloadRecord(existingCrons.get('firstObject'));
-    }
+    if (!branches) { return []; }
 
-    const cron = store.createRecord('cron', {
-      branch,
-      interval: this.get('selectedInterval') || 'monthly',
-      dont_run_if_recent_build_exists: this.get('selectedOption') || false
+    branches.forEach((branch) => {
+      branch.set('disabled', branchNames.indexOf(branch.name) > -1);
     });
+    return branches;
+  }),
 
-    this.reset();
+  actions: {
+    handleSubmit() {
+      const store = this.get('store');
 
-    yield cron.save();
-  }).drop(),
+      const cron = store.createRecord('cron', {
+        branch: this.get('selectedBranch'),
+        interval: this.get('selectedInterval'),
+        dont_run_if_recent_build_exists: this.get('selectedOption')
+      });
 
-  intervals: ['monthly', 'weekly', 'daily'],
+      this.reset();
+      cron.save();
+    },
 
-  options: ['Always run', 'Do not run if there has been a build in the last 24h']
+    onQueryChange(query) {
+      this.set('branchesFetcher.searchQuery', query);
+      this.get('branchesFetcher.performSearchRequest').cancelAll();
+      this.get('branchesFetcher.showSearchResults').perform();
```

Comment:
As per previous comments, `query` should be passed as an argument to the task instance via `perform` and you can do it directly in template:
```
{{field.select oninput=(perform 'performSearchRequest') }}
```